operator/api/v1beta1/helmrelease_types.go

```diff
@@ -26,6 +26,20 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type HelmReleaseStatusReason string
+
+const (
+	ReasonDeleteError     HelmReleaseStatusReason = "DeleteError"
+	ReasonDeleteSuccess   HelmReleaseStatusReason = "DeleteSuccess"
+	ReasonInstallError    HelmReleaseStatusReason = "InstallError"
+	ReasonInstallSuccess  HelmReleaseStatusReason = "InstallSuccess"
+	ReasonReconcileError  HelmReleaseStatusReason = "ReconcileError"
```

Comment:
What's this one?

Reply:
Internal controller/reconciler error. It might be useful for handling "impossible" errors, like covering the branch where a `obj.(*HelmRelease)` assertion fails, which should be impossible since this controller only listens for the HelmRelease API.

Maybe it's unnecessary, I just included all the possible reasons for a state change I could think of. I'd remove it if you think YAGNI, easy to add back if we want it again.

Reply:
I think YAGNI for now, I'd prefer we were pretty explicit here.